tests/unit/preprocessor/_multimodel/test_multimodel.py

```diff
@@ -921,3 +969,79 @@ def test_preserve_equal_coordinates():
     assert stat_cube.coord('x').standard_name is None
     assert stat_cube.coord('x').long_name is None
     assert stat_cube.coord('x').attributes == {}
+
+
+def test_arbitrary_dims_5d(cubes_5d):
+    """Test ``multi_model_statistics`` with 5D cubes."""
+    stat_cubes = multi_model_statistics(
+        cubes_5d,
+        span='overlap',
+        statistics=['sum'],
+    )
+    assert len(stat_cubes) == 1
+    assert 'sum' in stat_cubes
+    stat_cube = stat_cubes['sum']
+    assert stat_cube.shape == (1, 1, 1, 1, 1)
+    assert_array_allclose(
```

Comment:
why are we using allclose here and not an exact match?

Reply:
I think for floats this generally safer. Also, this (custom) function defined at the beginning of the module checks for masks, which is also nice to have :+1:

Reply:
OK - that does indeed do the masked check (I think I put it in there some 47 years ago :laughing: ) but the thing about floats is still bothering me - we should have machine-precision results, but let's deal with it if it hits us